esp/esdllib/esdl_script.cpp

```diff
@@ -328,42 +666,86 @@ class CEsdlTransformOperationSetValue : public CEsdlTransformOperationBase
         }
         return m_target.str();
     }
-    virtual bool doSet(IXpathContext * xpathContext, IPropertyTree *tree, const char *value)
+    virtual bool doSet(IXpathContext * sourceContext, IXpathContext *targetContext, const char *value)
     {
         StringBuffer xpath;
-        const char *target = getTargetPath(xpathContext, xpath);
-        ensurePTree(tree, target);
-        tree->setProp(target, value);
+        const char *target = getTargetPath(sourceContext, xpath);
+        targetContext->ensureSetValue(target, value, m_required);
         return true;
     }
 };
 
-class CEsdlTransformOperationRenameNode : public CEsdlTransformOperationBase
+class CEsdlTransformOperationNamespace : public CEsdlTransformOperationWithoutChildren
+{
+protected:
+    StringAttr m_prefix;
+    StringAttr m_uri;
+    bool m_current = false;
+
+public:
+    CEsdlTransformOperationNamespace(XmlPullParser &xpp, StartTag &stag, const StringBuffer &prefix) : CEsdlTransformOperationWithoutChildren(xpp, stag, prefix)
+    {
+        if (m_traceName.isEmpty())
+            m_traceName.set(stag.getValue("name"));
```

Comment:
Does the namespace operation really use @name? If not perhaps @prefix or @uri would be better fallbacks for the traceName.

Reply:
Yeah even the original use of name was a bit weird.  I'll remove it or use one of the other values for now.